### PR TITLE
fix(core): Fix ssl still active after change keystore in SSLServiceManager

### PR DESCRIPTION
**Pull request by salvatore-coppola**
Now in DataServiceImpl shutdownAutoConnectStrategy is invoked org.eclipse.kura.data.transport.listener.DataTransportListener.onConfigurationUpdated(boolean).

Lowered to 4 (lesser than the minimum RECONNECTION_MIN_DELAY) the sum of timeout and quiesce time of org.eclipse.paho.client.mqttv3.MqttAsyncClient.disconnectForcibly.

### CODE REVIEW

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java [hunk 1]
```diff
@@ -67,6 +67,8 @@
 public class DataServiceImpl implements DataService, DataTransportListener, ConfigurableComponent,
         CloudConnectionStatusComponent, CriticalComponent, AutoConnectStrategy.ConnectionManager {
 
+    private static final int RECONNECTION_MIN_DELAY = 5;
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java]: Does the workaround still work if we set this to 1 second? In this way we can be still consistent with the description of the connect.retry-interval configuration parameter, which states that the minimum value is 1 second.

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java [hunk 2]
```diff
@@ -1,12 +1,12 @@
 /*******************************************************************************
  * Copyright (c) 2011, 2022 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java]: Please update end year to 2023

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/transport/mqtt/MqttDataTransport.java [hunk 3]
```diff
@@ -1,12 +1,12 @@
 /*******************************************************************************
- * Copyright (c) 2011, 2021 Eurotech and/or its affiliates and others
- * 
+ * Copyright (c) 2011, 2022 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/transport/mqtt/MqttDataTransport.java]: Please update end year to 2023

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/transport/mqtt/MqttDataTransport.java [hunk 4]
```diff
@@ -99,6 +99,9 @@ public class MqttDataTransport implements DataTransportService, MqttCallback, Co
     private static final String TOPIC_ACCOUNT_NAME_CTX_NAME = "account-name";
     private static final String TOPIC_DEVICE_ID_CTX_NAME = "client-id";
 
+    private static final long MQTT_QUIESCE_TIMEOUT = 2;
+    private static final long MQTT_DISCONNECT_TIMEOUT = 2;
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/transport/mqtt/MqttDataTransport.java]: Can you please try to verify if issue https://github.com/eclipse/kura/issues/3071 occurs with the new settings?

File: kura/test/org.eclipse.kura.core.test/src/test/java/org/eclipse/kura/core/data/DataServiceImplTest.java [hunk 5]
```diff
@@ -387,7 +387,7 @@ public void testActivateAndFailConnecting()
         }
 
         // initial checkin + 3 * authentication + (other mqtt + 3)
-        assertEquals(8, count.intValue());
+        assertEquals(4, count.intValue());
```
Comment (nicolatimeus) [inline kura/test/org.eclipse.kura.core.test/src/test/java/org/eclipse/kura/core/data/DataServiceImplTest.java]: Please update copyright headers

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/transport/mqtt/MqttDataTransport.java (hunk 4 above)
Comment (salvatore-coppola) [reply kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/transport/mqtt/MqttDataTransport.java]: Checked

**Review by nicolatimeus — APPROVED** [review]

**Comment by nicolatimeus** [comment]
@MMaiero this PR decreases the timeout of disconnectForcibly that we introduced for fixing #3071. It speeds up the disconnection process and it seems that the thread leak cannot be reproduced. Is it ok for you to merge this? Were there specific reasons why we did not specify a lower timeout when we originally made the fix?

**Comment by MMaiero** [comment]
I have to admit that I don't remember the rationale for not setting a timeout.
Let's merge it and continue to verify during QA that everything works.

**Comment by nicolatimeus** [comment]
There seems to be failures in a DataService test

**Review by MMaiero — APPROVED** [review]

**Pull request merged**